docs/content/guides/dagster-pipes/integrating-databricks-with-dagster-pipes.mdx

```diff
@@ -1,6 +1,305 @@
+
+     - `libraries` - An object containing the libraries required to run the specified Python file (`spark_python_task.python_file`). This object must include `dagster-pipes`; for example:
+
+       ```python
+       "libraries": [
+          {
+          "pypi": {
+           "package": "dagster-pipes"
+          }
+         },
+       ],
+       ...
+       ```
+
+     - `spark_python_task` - An object specifying the Python file the job should run, which contains the following properties:
+       - `python_file` - The URI of the Python file to run, located in DBFS.
+       - `source` - TODO - is this required?
+
+### Step 1.2: Create the Databricks job
+
+In this step, we’ll create the Databricks job and pass in the task specification, as well as our `context` data. The Databricks resource used by the asset contains a `run` method. This method synchronously executes the Databricks job using the task specification you created in [the previous step](#step-11-define-a-dagster-asset).
+
+When the job is executed, Dagster Pipes will return a <PyObject module="dagster_pipes" object="PipesClientCompletedInvocation" /> object. This object contains a `get_results` method, which you can use to access the results reported by the Databricks job.
+
+In the body of the `databricks_asset`, add the following:
+
+```python
+# dagster_databricks_pipes.py
+
+return pipes_databricks.run(
+    task=task,
+    context=context,
+  ).get_results()
+```
+
+Let’s take a look at what this code does:
+
+- Submit the job to the Databricks resource (`pipes_databricks`) using the `run` method
+- Provide the `task` specification and asset `context` to the Databricks job
+- Use `get_results` to access the results of the execution
+- Lastly, return the results of the Databricks job
+
+### Step 1.3: Define the Databricks Pipes resource
+
+The [`dagster-databricks`](/\_apidocs/libraries/dagster-databricks) library provides a <PyObject module="dagster_databricks" object="PipesDatabricksClient" />, which is a pre-built Dagster resource that allows you to quickly get Pipes working with your Databricks workspace.
+
+1. In the imports section of `dagster_databricks_pipes.py`, add the following:
+
+   ```python
+   # dagster_databricks_pipes.py
+
+   from databricks.sdk import WorkspaceClient
+   ```
+
+2. Next, copy and paste the following to the bottom of `dagster_databricks_pipes.py`:
+
+   ```python
+   # dagster_databricks_pipes.py
+
+   pipes_databricks_resource = PipesDatabricksClient(
+     client = WorkspaceClient(
+      host=EnvVar("DATABRICKS_HOST"),
+       token=EnvVar("DATABRICKS_TOKEN")
+     )
+   )
+   ```
+
+In this example, we created a `pipes_databricks_resource`. The <PyObject module="dagster_databricks" object="PipesDatabricksClient" /> accepts a `client` argument, which provides the information required to connect to a Databricks workspace by using `WorkspaceClient`.
```

Comment:
"We created a `PipesDatabricksClient` object bound to the resource key `pipes_databricks_resource`."

BTW `pipes_databricks_resource` should be changed to `pipes_client`, per previous conversation with Nick this is the convention we're going with.

Reply:
> BTW pipes_databricks_resource should be changed to pipes_client, per previous conversation with Nick this is the convention we're going with.

Can I ask why this should change when it's user-defined? Couldn't a user have more than one type of Pipe client, and `pipes_client` would cause issues?